go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+		/*{
+			desc: 	"subnet IP input is nil",
+			inputGwIP: ovntest.MustParseIP("192.168.0.1"),
+			errExp: true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteListFiltered", []string{"int", "*netlink.Route", "uint64"}, []interface{}{[]netlink.Route{}, fmt.Errorf("mock error")}},
+			},
+		},*/
+		{
+			desc:        "tests code path where route GW IP DOES NOT MATCH with input GW IP",
+			inputGwIP:   ovntest.MustParseIP("192.168.0.1"),
+			inputSubnet: ovntest.MustParseIPNet("192.168.0.0/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteListFiltered", []string{"int", "*netlink.Route", "uint64"}, []interface{}{[]netlink.Route{
+					{Gw: ovntest.MustParseIP("192.168.1.1")},
+				}, nil}},
+			},
+		},
+		{
+			desc:        "tests code path where route GW IP MATCHES with input GW IP",
+			inputGwIP:   ovntest.MustParseIP("192.168.0.1"),
+			inputSubnet: ovntest.MustParseIPNet("192.168.0.0/24"),
+			outBoolFlag: true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteListFiltered", []string{"int", "*netlink.Route", "uint64"}, []interface{}{[]netlink.Route{
+					{Gw: ovntest.MustParseIP("192.168.0.1")},
+				}, nil}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			flag, err := LinkRouteExists(tc.inputLink, tc.inputGwIP, tc.inputSubnet)
+			t.Log(flag, err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
+			if tc.outBoolFlag {
+				assert.True(t, flag)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLinkNeighAdd(t *testing.T) {
+	mockNetLinkLibOperations := new(mocks.NetLinkLibOperations)
+	mockLink := new(netlink_mocks.Link)
+	// below is defined in net_linux.go
+	netLinkOperations = mockNetLinkLibOperations
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputNeigIP              net.IP
+		inputMacAddr             net.HardwareAddr
+		errExp                   bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+		onRetArgsLinkIfaceOpers  []onCallReturnArgs
+	}{
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "test code path with link argument set to nil",
+			errExp: true,
+		},*/
+		// NOTE: since, we dont validate function arguments in the funciton body, a nil value passed for neighIP and neighMac is sufficient
+		{
+			desc:      "test code path where adding neighbor returns an error",
+			inputLink: mockLink,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
```

Comment:
are you missing `errExp: true`?

Reply:
I believe so, let me confirm and fix accordingly